workspaces/spectacle/src/helpers.ts

```diff
@@ -84,13 +83,14 @@ export function buildEndpointChanges(
   shapeQueries: shapes.GraphQueries,
   sinceBatchCommitId?: string
 ): EndpointChanges {
+  // Sorted in reverse order - the latest change takes precedence
   const sortedBatchCommits = endpointQueries
     .listNodesByType(endpoints.NodeType.BatchCommit)
     .results.sort((a: any, b: any) => {
       return a.result.data.createdAt < b.result.data.createdAt ? 1 : -1;
-    });
+    }) as endpoints.BatchCommitNodeWrapper[];
```

Comment:
I think there might be some "cool" things we can do with typescript generics such that `listNodesByType` returns the actual node type instead of the generic `NodeWrapper`. Haven't gotten to dig into this though

Reply:
yep most of the graph queries should be able to produce more meaningful types, except the arbitrary neighbors/descendants ones, but I never got around to it